sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2406,6 +2407,15 @@ class Analyzer(override val catalogManager: CatalogManager) extends RuleExecutor
     }
   }
 
+  /** Helper method to resolve function calls within expressions of TableSpec options. */
+  def ResolveTableSpecOption(expression: Expression): Expression = {
```

Comment:
If we manage to transform UnresolvedTableSpec in the method `mapExpressions`, probably we don't need this method.

Reply:
👍 I was able to remove this entirely, and `ResolveTableSpecOption` is no longer a separate analyzer rule. Instead, `ResolveSessionCatalog` now simply just resolves the `TableSpec` when needed directly.